jwt.js

```diff
@@ -240,15 +240,10 @@ function fastifyJwt (fastify, options, next) {
       if (!token) {
         throw new BadRequestError()
       }
-    } else if ((request.headers && request.headers.authorization) && (!onlyCookie)) {
+    } else if (request.headers?.authorization && /^Bearer\s/i.test(request.headers.authorization) && !onlyCookie) {
```

Comment:
Unrelated but affects performance: why don't we place `onlyCookie` in the beginning?

```suggestion
    } else if (!onlyCookie && request.headers?.authorization && /^Bearer\s/i.test(request.headers.authorization)) {
```

Reply:
@cberescu can you do that as well?

If we set `onlyCookie = true`, we don't want to keep calling `RegExp.prototype.test` if someone for some reason repeatedly sends the auth header

Reply:
As long as they’re before the method call, it should be fine

> do we really need the ?.

I don’t think so